### ISSUE TITLE
(Doc) Need to have specific quarkus example for https://quarkus.io/guides/security-openid-connect-client-reference

### ISSUE THREAD

**Issue by orivat** [quarkusio/quarkus#39529]
### Describe the bug

1. quarkus-quickstarts/security-openid-connect-client-quickstart example is documented within Quarkus guide: https://quarkus.io/guides/security-openid-connect-client

2. There is also another guide https://quarkus.io/guides/security-openid-connect-client-reference,  but there is no genuine qurakus quckstart example corresponding to this doc URL 

### Expected behavior

This guide should be linked towards a new genuine quiarkus quickstart example

### Actual behavior

As long as it is not linked with genuine example, it will be very difficult to read and follow this page https://quarkus.io/guides/security-openid-connect-client-reference

### Output of `java -version`

17

### Quarkus version or git rev

3.8

**Comment by sberyozkin** [comment]
This second guide is a reference doc, we have a proper quickstart corresponding to the tutorial doc you listed in the 1st position. 

The reference doc lists so many options and variations that it is impossible to capture them all in an easy getting started with demo.

@orivat We can add more quickstarts but there should be specific cases which are not easy to figure out how to address feom the current quickstart. 

What I can do in this issue is to update the current quickstart to show the use of OidcClient created programmatically,  s8nce we updated the actual code

**Comment by orivat** [comment]
@sberyozkin

We definitely need to improve the way it is described.  Feel free to improve it as what you think it is the most the appropriate.

Other point:
It sounds like tokenHelper is never referenced within the quickstart (although it is also something imprtant). It might be good tohave also something for tokenhelper.

**Comment by sberyozkin** [comment]
@orivat Right, `TokensHelper` was given the coverage in #39367 but I'll also update the quickstart doc to explain when it can be needed. Until the recent discussion, I thought it could only be of the internal help to the filters, but now I know it can be useful when the OidcClient is used directly
